### PR TITLE
[RLlib] CRR documentation

### PR DESCRIPTION
**Pull request by kouroshHakha**
## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by christy — CHANGES_REQUESTED** [review]
Nice work, Kourosh!

Need to also add link refs to crr from:
- Github readme: https://github.com/ray-project/ray/blob/master/rllib/README.rst
- RLlib docs index > Algorithms pull-down: https://docs.ray.io/en/master/rllib/index.html 

New Algo Org:
Offline RL (put Off-policy Eval here): **CRR**, BC, MARWIL, CQL, OPE ("importance sampling" and "weighted importance sampling")
Model-free RL (split into off/on policy):
On-policy: APPO, PPO, IMPALA, A2C, A3C, PG
Off-policy: DQN, DDPG, TD3, SAC
Model-based RL: MBMPO, Dreamer
Derivative-free Algos: ES & ARS
RecSys and Bandits (or separate categories): LinUCB, LinTS, SlateQ
Multi-agent: QMIX, AlphaStar, MA-DDPG

**Review by christy — APPROVED** [review]

**Review by avnishn — APPROVED** [review]

File: doc/source/rllib/rllib-algorithms.rst [hunk 1]
```diff
@@ -634,6 +635,28 @@ Tuned examples: `HalfCheetah Random <https://github.com/ray-project/ray/blob/mas
...
+.. _crr:
+
+Critic Regularized Regression (CRR)
+-----------------------------------
+|pytorch|
+`[paper] <https://arxiv.org/abs/2006.15134>`__ `[implementation] <https://github.com/ray-project/ray/blob/master/rllib/algorithms/crr/crr.py>`__
+
+CRR is another offline RL algorithm based on Q-learning that can learn from an offline experience replay.
+The challenge in applying existing Q-learning algorithms to offline RL is the problem of overestimating Q-function, as well as lack of exploration beyond the observed data.
+The later become increasingly important during bootstrapping the bellman equation, where the Q-function queried for next state does not have support in the observed data.
```
Comment (sven1977) [inline doc/source/rllib/rllib-algorithms.rst:648]:
```suggestion
The latter becomes increasingly important during bootstrapping in the bellman equation, where the Q-function queried for the next state's Q-value(s) does not have support in the observed data.
```

File: doc/source/rllib/rllib-algorithms.rst [hunk 2]
```diff
@@ -634,6 +635,28 @@ Tuned examples: `HalfCheetah Random <https://github.com/ray-project/ray/blob/mas
...
+To mitigate these issues, CRR, implements a simple and yet powerful idea of value-filtered regression.
```
Comment (sven1977) [inline doc/source/rllib/rllib-algorithms.rst:649]:
```suggestion
To mitigate these issues, CRR implements a simple and yet powerful idea of "value-filtered regression".
```

File: doc/source/rllib/rllib-algorithms.rst [hunk 3]
```diff
@@ -634,6 +635,28 @@ Tuned examples: `HalfCheetah Random <https://github.com/ray-project/ray/blob/mas
...
+Basically the key idea is to use a learned critic to filter-out the non-promising transitions from the replay dataset. For more details, please refer to the paper.
```
Comment (sven1977) [inline doc/source/rllib/rllib-algorithms.rst:650]:
```suggestion
The key idea is to use a learned critic to filter-out the non-promising transitions from the replay dataset. For more details, please refer to the paper (see link above).
```

**Review by sven1977 — APPROVED** [review]
Great work @kouroshHakha , thanks for the PR.

**Pull request merged**